### PR TITLE
 canvas ribbon

### PR DESCRIPTION
**Pull request by warmqing**
added:canvas_ribbon.js
image

### CODE REVIEW

File: layout/includes/additional-js.pug [hunk 1]
```diff
@@ -8,6 +8,8 @@ if (theme.katex && theme.katex.enable)
     include ./third-party/katex.pug
 if (theme.local_search && theme.local_search.enable)
   script(src=url_for('/js/search/local-search.js'))
+if (theme.canvas_ribbon && theme.canvas_ribbon.enable)
+  include ./third-party/canvas_ribbon.pug
```
Comment (Molunerfinn) [inline layout/includes/additional-js.pug]: change the `canvas_ribbon.pug` to `canvas-ribbon.pug` would be better

File: source/js/canvas_ribbon.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/**
```
Comment (Molunerfinn) [inline source/js/canvas_ribbon.js]: change the `canvas_ribbon.js` to `canvas-ribbon.js` would be better

File: layout/includes/third-party/canvas_ribbon.pug [hunk 3]
```diff
@@ -0,0 +1,2 @@
+script(id="ribbon" src=url_for('/js/canvas_ribbon.js') size=theme.canvas_ribbon.size
```
Comment (Molunerfinn) [inline layout/includes/third-party/canvas_ribbon.pug]: change the `canvas_ribbon.js` to `canvas-ribbon.js` would be better

**Comment by Molunerfinn** [comment]
Nice, once you change the names of these files, I will merge this PR!

**Comment by jerryc127** [comment]
建议把canvas-ribbon换成这个https://github.com/hustcc/ribbon.js
canvas-ribbon在高分屏下有bug,反馈到github了，但好像不再维护了。
https://github.com/hustcc/ribbon.js暂时没发现问题

**Comment by Molunerfinn** [comment]
@jerryc127 收到，我会测试一下

**Pull request merged**